src/ion/moves.rs

```diff
@@ -288,24 +403,7 @@ impl<'a, F: Function> Env<'a, F> {
                         break;
                     }
                     trace!("examining block with end in range: block{}", block.index());
-                    for &succ in self.func.block_succs(block) {
-                        trace!(
-                            " -> has succ block {} with entry {:?}",
-                            succ.index(),
-                            self.cfginfo.block_entry[succ.index()]
-                        );
-                        if range.contains_point(self.cfginfo.block_entry[succ.index()]) {
-                            continue;
-                        }
-                        trace!(" -> out of this range, requires half-move if live");
-                        if self.is_live_in(succ, vreg) {
-                            trace!("  -> live at input to succ, adding halfmove");
-                            half_moves.push(HalfMove {
-                                key: half_move_key(block, succ, vreg, HalfMoveKind::Source),
-                                alloc,
-                            });
-                        }
-                    }
+                    inter_block_sources.insert(block, alloc);
```

Comment:
Note from live review: let's consider the overlap case here and perhaps adopt a simple heuristic to keep one over the other (e.g., if register already, don't replace with stack, but not vice-versa).